plugins/runinterminal.lua

```diff
@@ -0,0 +1,101 @@
+local core = require "core"
+local common = require "core.common"
+local command = require "core.command"
+local config = require "core.config"
+local keymap = require "core.keymap"
+
+local run = {}
+
+-- return a function which will run the current doc
+-- as a script for the given language
+function run.lang(lang)
+  return function()
+    local doc = core.active_view.doc
+    if doc.filename then
+      doc:save()
+    else
+      core.error("Cannot run an unsaved file")
```

Comment:
unnamed instead of unsaved?
just nitpicking while reviewing :)

Reply:
Yeah, lite labels such files as "unsaved" so I did the same